### PR TITLE
[RPA-960]: Set cell refactor

### PR DESCRIPTION
**Pull request by Klau-Sam**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-desktop/src/action/sharepoint/excel/sharepoint-excel.action-handler.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default class SharepointExcelActionHandler extends StatelessActionHandler
     async setCell(
         input: SharepointTypes.SharepointSetExcelCellActionInput
     ): Promise<SharepointTypes.SharepointExcelSetCellActionOutput> {
-        return this.excelService.setCell(input.cell, input.content);
+        this.excelService.setCell(input.cell, input.value);
     }
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/sharepoint/excel/sharepoint-excel.action-handler.ts]: if no response then adjust return type

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 2]
```diff
@@ -73,6 +70,7 @@ export class ExcelService {
     }
     // https://learn.microsoft.com/en-us/graph/api/worksheet-cell?view=graph-rest-1.0&tabs=http
     async getCell(cellCoordinates: WorkbookCellCoordinates): Promise<ExcelCellValue> {
+        this.checkSession();
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: session should be checked at action handler

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/process.json [hunk 3]
```diff
@@ -308,9 +308,9 @@
     "Process.Details.Modeler.Actions.SharePointExcel.GetRange.StartCell.Info": "Adres komórki początkowej (np. A1)",
     "Process.Details.Modeler.Actions.SharePointExcel.GetRange.EndCell": "Komórka końcowa",
     "Process.Details.Modeler.Actions.SharePointExcel.GetRange.EndCell.Info": "Adres komórki końcowej (np. H5)",
-    "Process.Details.Modeler.Actions.SharepointExcel.SetCell.Label": "Dodaj zawartość do komórki",
-    "Process.Details.Modeler.Actions.SharePointExcel.SetCell.Content": "Zawartość komórki",
-    "Process.Details.Modeler.Actions.SharePointExcel.SetCell.Cell": "Adres komórki",
+    "Process.Details.Modeler.Actions.SharepointExcel.SetCell.Label": "Ustaw wartość komórki",
```
Comment (bhszewczyk) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/process.json]: Suggestion: "Ustaw komórkę" (there is "Pobierz komórkę", "Pobierz komórki", and English "Set cell")

File: runbotics/runbotics-orchestrator-ui/src/src-app/Actions/sharepointExcel.actions.ts [hunk 4]
```diff
@@ -173,49 +167,33 @@ const getSharepointExcelActions: () => Record<string, IBpmnAction> = () => ({
...
                         type: 'object',
                         properties: {
-                            content: {
-                                title: translate('Process.Details.Modeler.Actions.SharePointExcel.SetCell.Content'),
-                                type: 'string',
-                            },
                             cell: {
                                 title: translate('Process.Details.Modeler.Actions.SharePointExcel.SetCell.Cell'),
                                 type: 'string',
                             },
```
Comment (bhszewczyk) [inline runbotics/runbotics-orchestrator-ui/src/src-app/Actions/sharepointExcel.actions.ts]: you could add a pattern to validate the input (regex already exists: ActionRegex.EXCEL_CELL_ADDRESS), now if the user types only A the input is correct an the action is successfully completed. I think we should capture such incorrect input and throw an error

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts [hunk 5]
```diff
@@ -2,32 +2,26 @@ export type SessionIdentifier = string;
...
     list?: string
 }
 
-export interface SharePointSessionInput {
-    platform: Platform,
-    sessionIdentifier: SessionIdentifier,
-    worksheetIdentifier: WorksheetIdentifier,
-    siteRelativePath?: string,
-    list?: string
+export interface Session extends OneDriveSessionInput {
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts]: imo Session should be top-level type and other types of sessions should inherit from it

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts (hunk 5 above)
Comment (Klau-Sam) [reply runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts]: but in this case the hierarchy is reversed. "global" session object includes OneDrive and SharePoint session

**Pull request closed without merging**